test/hotspot/jtreg/runtime/cds/appcds/jcmd/JCmdTestFileSafety.java

```diff
@@ -76,17 +79,29 @@ static void test() throws Exception {
         outputDirFile.setWritable(true);
         String localFileName = subDir + File.separator + "MyStaticDump.jsa";
         setIsStatic(true/*static*/);
+        File fileLocal = new File(localFileName);
+        if (fileLocal.exists()) {
+            fileLocal.delete();
+        }
         // Set target dir not writable, do static dump
         print2ln(test_count++ + " Set target dir not writable, do static dump");
         setKeepArchive(true);
         app = createLingeredApp("-cp", allJars);
         pid = app.getPid();
         test(localFileName, pid, noBoot,  EXPECT_PASS);
+        checkFileExistence(localFileName, true/*exist*/);
+        FileTime ft1 = Files.getLastModifiedTime(Paths.get(localFileName));
         outputDirFile.setWritable(false);
         test(localFileName, pid, noBoot,  EXPECT_FAIL);
-        outputDirFile.setWritable(true);
-        checkFileExistence(localFileName, true/*exist*/);
+        FileTime ft2 = Files.getLastModifiedTime(Paths.get(localFileName));
+        if (!ft2.equals(ft1)) {
+            throw new RuntimeException("Archive file " + localFileName + " should not be updated");
+        }
 
+        if (fileLocal.exists()) {
+            fileLocal.delete();
+        }
+        outputDirFile.setWritable(true);
```

Comment:
Since you've added the `setWritable(true)` here, do you still need the existing one at line 116?

Reply:
It can be removed, but better keep it --- the testing result of last step is deleted just in case it could affect next test. I would like remove the result file anyway. Since the file name keep changing, I will update with a function with the file name (String) to delete the file. Thanks.